nemo_curator/utils/distributed_utils.py

```diff
@@ -583,3 +585,29 @@ def performance_report_if(path=None, report_name="dask-profile.html"):
         return performance_report(os.path.join(path, report_name))
     else:
         return nullcontext()
+
+
+def seed_everything(seed: int = 42):
+    """
+    Function to set seed for random number generators for reproducibility.
+
+    Args:
+        seed: The seed value to use for random number generators. Default is 42.
+
+    Returns:
+        None
+    """
+    ## Imporing torch to help with context issues
+    import torch
+
+    # Set seed values for various random number generators
+    random.seed(seed)
+    os.environ["PYTHONHASHSEED"] = str(seed)
+    np.random.seed(seed)
+    torch.manual_seed(seed)
+    torch.cuda.manual_seed(seed)
```

Comment:
Will this work if there is no GPU?

Reply:
Resolved by https://github.com/NVIDIA/NeMo-Curator/pull/130/commits/f957b507d8d47e00cda0cf4bc64001e9f1546f4c